### PR TITLE
🚧 iMXRT10xxx and FreeRTOS Port

### PR DESCRIPTION
**Pull request by MateuszKlatecki**
## Description
- The first phase of porting nanoframework to FreeRTOS and iMX.RT10xx.
- Created cmake files. An image is being built with a running FreeRTOS system and a flashing LED diode and sending messages via the serial port.

## Motivation and Context
- Add support for mainstream FreeRTOS.
- Add new reference target (iMX.RT10xx) from a different vendor (NXP). 

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: CMake/Modules/FreeRTOS_iMXRT10xx_GCC_options.cmake [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2017 The nanoFramework project contributors
```
Comment (josesimoes) [inline CMake/Modules/FreeRTOS_iMXRT10xx_GCC_options.cmake]: please change this to 2019

File: CMake/Modules/FindFreeRTOS.cmake [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#
+# Copyright (c) 2017 The nanoFramework project contributors
```
Comment (josesimoes) [inline CMake/Modules/FindFreeRTOS.cmake]: please change this to 2019

File: CMake/Modules/FreeRTOS_iMXRT10xx_sources.cmake [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#
+# Copyright (c) 2017 The nanoFramework project contributors
```
Comment (josesimoes) [inline CMake/Modules/FreeRTOS_iMXRT10xx_sources.cmake]: please change this to 2019

File: CMake/toolchain.FreeRtos.GCC.cmake [hunk 4]
```diff
@@ -0,0 +1,171 @@
+INCLUDE(CMakeForceCompiler)
```
Comment (josesimoes) [inline CMake/toolchain.FreeRtos.GCC.cmake]: missing copyright header

File: CMake/toolchain.FreeRtos.GCC.cmake [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+# if( DEFINED CMAKE_CROSSCOMPILING )
```
Comment (josesimoes) [inline CMake/toolchain.FreeRtos.GCC.cmake]: I get it that this commented code is a leftover from the original toolchain file. 
We should look into tidy the code so this doesn't belong here. Please remove it.

File: targets/FreeRTOS/NXP_MIMXRT1060_EVK/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,155 @@
+#
+# Copyright (c) 2017 The nanoFramework project contributors
```
Comment (josesimoes) [inline targets/FreeRTOS/NXP_MIMXRT1060_EVK/CMakeLists.txt]: please change this to 2019

File: targets/FreeRTOS/NXP_MIMXRT1060_EVK/common/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2017 The nanoFramework project contributors
```
Comment (josesimoes) [inline targets/FreeRTOS/NXP_MIMXRT1060_EVK/common/CMakeLists.txt]: guess what? 2019... 😉

File: targets/FreeRTOS/NXP_MIMXRT1060_EVK/common/board/board.c [hunk 8]
```diff
@@ -0,0 +1,161 @@
+/*
```
Comment (josesimoes) [inline targets/FreeRTOS/NXP_MIMXRT1060_EVK/common/board/board.c]: The license header should be something like this:
//
// Copyright (c) 2017 The nanoFramework project contributors
// Portions Copyright 2018 NXP. All rights reserved.
// See LICENSE file in the project root for full license information.
//

File: targets/FreeRTOS/NXP_MIMXRT1060_EVK/common/board/board.h [hunk 9]
```diff
@@ -0,0 +1,225 @@
+/*
```
Comment (josesimoes) [inline targets/FreeRTOS/NXP_MIMXRT1060_EVK/common/board/board.h]: The license header should be something like this:
//
// Copyright (c) 2017 The nanoFramework project contributors
// Portions Copyright 2018 NXP. All rights reserved.
// See LICENSE file in the project root for full license information.
//

File: targets/FreeRTOS/NXP_MIMXRT1060_EVK/nanoBooter/MIMXRT10xx.ld [hunk 10]
```diff
@@ -0,0 +1,297 @@
+/*
+//
+// Copyright (c) 2017 The nanoFramework project contributors
+// Portions Copyright (c) 2006..2015 Giovanni Di Sirio. All rights reserved.
```
Comment (josesimoes) [inline targets/FreeRTOS/NXP_MIMXRT1060_EVK/nanoBooter/MIMXRT10xx.ld]: this line doesn't seem correct.... maybe NXP bla, bla?...

**Comment by MateuszKlatecki** [comment]
I added and/or corrected copyright notes
CMSIS is now pulled from a repo

**Pull request merged**